pylint/checkers/spelling.py

```diff
@@ -185,6 +185,7 @@ def _next(self):
 
 
 CODE_FLANKED_IN_BACKTICK_REGEX = re.compile(r"(\s|^)(`{1,2})([^`]+)(\2)([^`]|$)")
+MYPY_IGNORE_DIRECTIVE_RULE_REGEX = re.compile(r"(\s|^)(type\: ignore\[)([^\]]+)(\].*)")
```

Comment:
```suggestion
MYPY_IGNORE_DIRECTIVE_RULE_REGEX = re.compile(r"(\s|^)(type\: ignore\[[^\]]+\])(.*)")
```
<img width="722" alt="Schermafbeelding 2022-05-05 om 21 17 30" src="https://user-images.githubusercontent.com/13665637/167009416-61fc5313-06d7-442d-bde5-79ec5f159a4e.png">

This creates the following three groups. I don't think we want to check anything in between `type` until the last bracket.